sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -710,6 +722,8 @@ class GpuDynamicPartitionDataConcurrentWriter(
     }
   }
 
+  override protected val reportSingleWriter: Boolean = false
```

Comment:
why do we not want to report a single writer?

Reply:
Because the concurrent write has its own writers and report the writers size separately in the `write` method.

Reply:
Here is the code snippet.
```
    // Split the batch and cache the result, along with opening the writers.
    splitBatchToCacheAndClose(cb)
    // Update the maxWriterNumber metric since the writers are already opened and cached.
    statsTrackers.foreach(_.writersNumber(concurrentWriters.size))  <====== Report the writers number here.
    // Write the cached batches
    val writeFunc: (WriterIndex, WriterStatusWithBatches) => Unit =
```